controller/controller.go

```diff
@@ -368,6 +379,13 @@ func (ctrl *ProvisionController) shouldProvision(claim *v1.PersistentVolumeClaim
 		return false
 	}
 
+	if failureCount, exists := ctrl.failedClaimsStats[claim.UID]; exists == true {
```

Comment:
it's possible for shouldProvision to read from the map and updateStats to write to the map at the same time. Can we protect it with a lock to be sure.